src/App.js

```diff
@@ -37,35 +49,110 @@ function AppNavbar() {
+        </Dropdown>
+        {/* <PWAInstallButton /> */}
       </Navbar>
     </>
   );
-}
+};
 
-function App() {
-  return (
-    <Router history={history}>
-      <div className="App">
-        <ScrollToTop />
-        <AppNavbar />
-        <div className="page">
-          <Switch>
-            <Route path="/update" component={SubmitPage} />
-            <Route path="/location" component={LocationSelectionPage} />
-            <Route path="/about" component={AboutPage} />
-            <Route path="/" component={Homepage} />
-          </Switch>
-        </div>
-        <footer className="m-0 p-0">
-          <div className="container py-4 text-center text-uppercase">
-            <Link to="/">Home</Link> · <Link to="/location">Add a store</Link> ·{" "}
-            <Link to="/about">About</Link>
+const AppContext = createContext();
+
+class App extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      language: this.getDefaultLanguage(),
+    };
+  }
+
+  setLanguage = language => {
+    this.persistLastSelectedLanguage(language);
+    this.setState({ language });
+  };
+
+  persistLastSelectedLanguage = language => {
+    if (window.localStorage) {
+      localStorage.setItem("lastSelectedLanguage", language);
+    }
+  };
+
+  getDefaultLanguage = () => {
+    if (window.localStorage) {
+      const language = localStorage.getItem("lastSelectedLanguage");
+      return language || AVAILABLE_LANGUAGES.ENGLISH;
+    }
+  };
+
+  getTranslations = () => {
+    return {
+      ...translations[AVAILABLE_LANGUAGES.ENGLISH],
+      ...translations[this.state.language],
+    };
+  };
+
+  render() {
+    const translations = this.getTranslations();
+    return (
+      <Router history={history}>
+        <AppContext.Provider
+          value={{
+            translations,
+          }}
+        >
+          <div className="App">
+            <ScrollToTop />
+            <AppNavbar
+              currentLanguage={this.state.language}
+              setLanguage={this.setLanguage}
+            />
+            <div className="page">
+              <Switch>
+                <Route path="/update" component={SubmitPage} />
+                <Route path="/location" component={LocationSelectionPage} />
+                <Route path="/about" component={AboutPage} />
+                <Route path="/" component={Homepage} />
+              </Switch>
+            </div>
+            <footer className="m-0 p-0">
+              <div className="container py-4 text-center text-uppercase">
+                <Link to="/">{translations.home}</Link> ·{" "}
+                <Link to="/location">{translations.add_store}</Link> ·{" "}
+                <Link to="/about">{translations.about}</Link>
+              </div>
+            </footer>
           </div>
-        </footer>
-      </div>
-    </Router>
-  );
+        </AppContext.Provider>
+      </Router>
+    );
+  }
+}
+
+export function withGlobalContext(Component) {
```

Comment:
i haven't use withGlobalContext before but it looks neat. i was setting up a redux store in #117 to manage current locations as props for the different pages. do you think global context is better?

Reply:
Yes. This can be used just like a redux store. Redux does have its benefits of being more explicit with dispatches, but for such small scale projects its boilerplate can be overkill.

Reply:
i agree. i'll remove the redux bits